linkedList.js

```diff
@@ -0,0 +1,372 @@
+class Node {
+  constructor(data) {
+    this._data = data;
+    this.nextNode = null;
+  }
+
+  getData() {
+    return this._data;
+  }
+
+
+}
+
+class LinkedList {
+  constructor() {
+    this.head = null;
+  }
+
+  length() {
+    var currentNode = this.head;
+    var length = 0;
+
+    if(!currentNode) {
+      return null;
+    }
+
+    while(currentNode) {
+      length++;
+      currentNode = currentNode.nextNode;
+    }
+    return length;
+  }
+
+  insert(value) {
+    var node = new Node(value);
+    var currentNode = this.head;
+
+    if(!currentNode) {
+      this.head = node;
+
+      return node;
+    }
+
```

Comment:
This looks good! Just as an additional optimization, your code will work even if you delete lines 38 through 42.